doc/plutus-core-spec/cardano/builtins6.tex

```diff
@@ -63,6 +63,36 @@ \subsubsection{Built-in functions}
             [x_{k+1}, \ldots, x_n]  & \text{if $1  \leq k \leq n-1$} \\ \relax
             []                     &\text{if $k \geq n$}\\
         \end{array}\right.$} &  & \\
+    \TT{lengthOfArray} 
+      & $[\forall a_\#, \arrayOf{a_\#}] \to \ty{integer}$ 
+      & $[] \mapsto 0$
+        \newline 
+        $[x_1] \mapsto 1$
+        \newline 
+        $[x_1,\ldots,x_n] \mapsto n$ 
+      &  
+      & \\
+    \TT{listToArray} 
+      & $[\forall a_\#, \listOf{a_\#}] \to \arrayOf{a_\#}$ 
+      & $[] \mapsto []$
+        \newline 
+        $[x_1] \mapsto [x_1]$
+        \newline 
+        $[x_1,\ldots,x_n] \mapsto [x_1,\ldots,x_n]$ 
```

Comment:
I don't know a better denotation. Maybe we should use a different syntax to denotate arrays? @effectfully please suggest, what would you do?

Reply:
This is how the whole table looks when rendered:
![image](https://github.com/user-attachments/assets/efdf853b-c56a-466f-8884-7208a169a0d0)

Reply:
This looks good, although I think you could miss out the two `[x₁] ↦ ...` cases since they're covered by `[x₁, ..., xₙ]` with n=1 (and arguably the `[]` case is covered by n=0, but that might be a bit confusing).

Also,
```
[x₁, ..., xₙ] ↦ [x₁, ..., xₙ]
```
might look a bit weird,but on page 33 we've already got `(x,y) ↦ (x,y)` for the denotation of `mkPairData`, so its maybe not such a big deal.

Reply:
> I think you could miss out the two [x₁] ↦ ... cases since they're covered by [x₁, ..., xₙ] with n=1 (and arguably the [] case is covered by n=0, but that might be a bit confusing).

Yeah, I wasn't sure if `[x₁, ..., xₙ] ↦ [x₁, ..., xₙ]` subsumes `[x₁] ↦ [x₁]` or even `[] ↦ []` so I went for more explicit notation. Happy to simplify it if you think that there is an ambiguity.

Reply:
![image](https://github.com/user-attachments/assets/883669e8-2d7e-407d-b5f2-e4b0de0d796e)